server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -639,4 +641,395 @@ private TrackingTool toTrackingTool(CRTrackingTool crTrackingTool) {
+        CRPluggableTask task = new CRPluggableTask(pluginConfiguration, configuration);
+        commonCRTaskMembers(task, pluggableTask);
+        return task;
+    }
+
+    public CRExecTask execTasktoCRExecTask(ExecTask task) {
+        CRExecTask crExecTask = new CRExecTask(task.getCommand());
+        crExecTask.setTimeout(task.getTimeout());
+        crExecTask.setWorkingDirectory(task.workingDirectory());
+        crExecTask.setArgs(task.getArgList().stream()
+                .map(Argument::getValue)
+                .collect(Collectors.toList())
+        );
+
+        commonCRTaskMembers(crExecTask, task);
+        return crExecTask;
+    }
+
+    public CRBuildTask buildTaskToCRBuildTask(BuildTask buildTask) {
+        CRBuildTask crBuildTask;
+        if (buildTask instanceof RakeTask) {
+            RakeTask rake = (RakeTask) buildTask;
+            crBuildTask = CRBuildTask.rake();
+        } else if (buildTask instanceof  AntTask) {
+            crBuildTask = CRBuildTask.ant();
+        } else if (buildTask instanceof NantTask) {
+            crBuildTask = CRBuildTask.nant(((NantTask) buildTask).getNantPath());
+        } else {
+            throw new RuntimeException(
+                    String.format("unknown type of build task '%s'", buildTask));
+        }
+        crBuildTask.setBuild_file(buildTask.getBuildFile());
+        crBuildTask.setTarget(buildTask.getTarget());
+        crBuildTask.setWorking_directory(buildTask.workingDirectory());
+        commonCRTaskMembers(crBuildTask, buildTask);
+        return crBuildTask;
+    }
+
+    private void commonCRTaskMembers(CRTask crTask, AbstractTask task) {
+        Task taskOnCancel = task.cancelTask();
+        if (taskOnCancel != null && !(taskOnCancel instanceof KillAllChildProcessTask))
+            crTask.setOn_cancel(taskToCRTask(taskOnCancel));
+        crTask.setRunIf(crRunIfs(task.runIfConfigs));
+    }
+
+    private CRRunIf crRunIfs(RunIfConfigs runIfs) {
+        if (runIfs == null || runIfs.isEmpty())
+            return CRRunIf.passed;
+       RunIfConfig runIf = runIfs.first();
+        if (runIf.equals(RunIfConfig.ANY)) {
+           return CRRunIf.any;
+        } else if (runIf.equals(RunIfConfig.PASSED)) {
+            return CRRunIf.passed;
+        } else if (runIf.equals(RunIfConfig.FAILED)) {
+            return CRRunIf.failed;
+        } else {
+            throw new RuntimeException(
+                    String.format("unknown run if condition '%s'", runIf));
+        }
+    }
+
+    private List<CRConfigurationProperty> configurationToCRConfiguration(Configuration config) {
+        ArrayList<CRConfigurationProperty> properties = new ArrayList();
+        if (config != null) {
+            for (ConfigurationProperty p : config) {
+                CRConfigurationProperty crProp = new CRConfigurationProperty(p.getKey().getName());
+                if (p.getValue() != null)
```

Comment:
Should check `isSecure`, instead of `getValue`. `getValue` decrypts on the fly and is not often `null`. `getConfigValue` might work, but it is probably more readable to check `isSecure`?